src/AutoMapper/Internal/TypePair.cs

```diff
@@ -7,10 +7,19 @@ public MapRequest(TypePair types) : this(types, types, MemberMap.Instance) { }
     public override int GetHashCode() => HashCode.Combine(RequestedTypes, RuntimeTypes);
 }
 [DebuggerDisplay("{SourceType.Name}, {DestinationType.Name}")]
-public readonly record struct TypePair(Type SourceType, Type DestinationType)
+public readonly record struct TypePair
 {
+    public TypePair(Type sourceType, Type destinationType)
+    {
+        SourceType = sourceType;
+        DestinationType = destinationType.IsByRef ? destinationType.GetElementType() : destinationType;
```

Comment:
No, what I meant, no change is ok here, in this class. Only constructor mapping code can change.